src/map.rs

```diff
@@ -43,32 +57,37 @@ impl Map {
         if self.paths.contains_key(&path) {
             panic!("oh no!")
         }
-        self.items.insert(item.code, vec![]);
+        let item = Rc::new(item);
+        self.items.insert(
+            item.code,
+            MapItemList {
+                values: vec![],
+                item: item.clone(),
```

Comment:
Use `Rc::clone(&item)` instead of `.clone()` to make it clear that we are cloning the `Rc`

Reply:
done